apps/src/Sounds.js

```diff
@@ -328,6 +372,67 @@ Sounds.prototype.playURL = function(url, playbackOptions) {
   }
 };
 
+/**
+ * @param {Promise} promise the promise that will resolve to the ArrayBuffer bytes to be played
+ * @param {object} playbackOptions config for the playing of the sound.
+ * @param {object} cacheParams optional parameters for caching the result of promise
+ */
+Sounds.prototype.addPromiseToSpeechQueue = function(
+  promise,
+  playbackOptions,
+  cacheParams = undefined
+) {
+  this.speechQueue.push({promise, playbackOptions, cacheParams});
+  this.checkSpeechQueue();
+};
+
+Sounds.prototype.checkSpeechQueue = async function() {
+  if (!this.speechPlaying && this.speechQueue.length > 0) {
```

Comment:
nit: we can remove 1 level of nesting if we invert this conditional and return early